register.lic

```diff
@@ -0,0 +1,76 @@
+=begin
+  Search a deed register for the specified crafting material.
+  Output all lines that match.
+=end
+
+custom_require.call(%w[common common-items drinfomon])
+
+class Register
+  include DRC
+  include DRCI
+
+  def initialize
+    settings = get_settings
+	@search_results = []
+
+    arg_definitions =
+      [
+        [
+          { name: 'query', regex: /^[A-z\s\-\']+$/i, description: 'crafting material to search for, use quotes for multiple words' }
+        ]
+      ]
+
+    args = parse_args(arg_definitions)
+
+    if args.query
+      stow_hands
+	  if get_item('deed register', get_settings.crafting_container)
+	    fput("turn my register to contents")
+        search(args.query)
+	  else
+	    echo "Can\'t find your deed register."
+		exit
+	  end
+      stow_hands
+	end
+	respond('   ' + "Results:")
+	if @search_results.any?
+	  display_results(@search_results)
+	else
+	  respond('   ' + "No matches found.")
+	end
+  end
+
+  def display_results(results)
+    results.each do |data|
+      respond("#{data}")
+    end
+  end
+
+  def search(query)
```

Comment:
You don't need to make this change, but I wanted to point out a style thing. You could have `search` return the results (instead of assigning them to `@search_results`) and then you don't need to manage an instance var.

Reply:
I see what you mean.  I can just call search and have it do all the printing that I currently have in the initialize section.  That would be cleaner.  Change made.